dlc-test/src/test/scala/org/bitcoins/dlc/DLCClientTest.scala

```diff
@@ -208,11 +208,88 @@ class DLCClientTest extends BitcoinSAsyncTest {
       HashType.sigHashAll
     )
 
+    def validateFees(
+        fundingTx: Transaction,
+        closingTx: Transaction): Assertion = {
+      val builder = dlcOffer.dlcTxBuilder
+      val finalizer = builder.fundingTxBuilder.fundingTxFinalizer
+      val expectedFundingFee =
+        finalizer.offerFundingFee + finalizer.acceptFundingFee
+      val expectedClosingFee =
+        finalizer.offerFutureFee + finalizer.acceptFutureFee
+
+      val fundingInput =
+        (builder.offerFundingInputs ++ builder.acceptFundingInputs)
+          .map(_.output.value)
+          .sum
+      val fundingOutput = fundingTx.outputs.map(_.value).sum
+      val actualFundingFee = fundingInput - fundingOutput
+
+      val closingInput = fundingTx.outputs.head.value
+      val closingOutput = closingTx.outputs.map(_.value).sum
+      val actualClosingFee = closingInput - closingOutput
+
+      def feeRateBounds(
+          tx: Transaction,
+          actualFee: CurrencyUnit,
+          missingOutputBytes: Long = 0): (Double, Double) = {
+        val vbytesLower = Math.ceil(tx.weight / 4.0) + missingOutputBytes
+        val vbytesUpper =
+          vbytesLower + 2 // Have to add 2 to account for two parties rounding up
+
+        // Upper VBytes => Lower fee rate
+        val lowerBound = actualFee.satoshis.toLong / vbytesUpper
+        // Lower VBytes => Upper fee rate
+        val upperBound = actualFee.satoshis.toLong / vbytesLower
+        (lowerBound, upperBound)
```

Comment:
what about rounding here

Reply:
not sure what you mean?

Reply:
(both returned values are Double's btw so shouldn't have a rounding issue?)

Reply:
You're doing division here with 2 longs, so there will be loss of precision. Maybe the `upperBound` should be in a `Math.ceil`?

Reply:
the denominators are Doubles not Longs (Math.ceil returns a double)